packages/jest/src/executors/jest/jest.impl.ts

```diff
@@ -2,23 +2,109 @@ import 'dotenv/config';
 import { runCLI } from 'jest';
 import { readConfig } from 'jest-config';
 import { utils as jestReporterUtils } from '@jest/reporters';
-import { makeEmptyAggregatedTestResult, addResult } from '@jest/test-result';
+import { addResult, makeEmptyAggregatedTestResult } from '@jest/test-result';
 import * as path from 'path';
+import { join } from 'path';
 import { JestExecutorOptions } from './schema';
 import { Config } from '@jest/types';
-import { ExecutorContext, TaskGraph } from '@nrwl/devkit';
-import { join } from 'path';
+import {
+  ExecutorContext,
+  offsetFromRoot,
+  ProjectGraphProjectNode,
+  readJsonFile,
+  TaskGraph,
+} from '@nrwl/devkit';
 import { getSummary } from './summary';
+import { readCachedProjectGraph } from '@nrwl/workspace/src/core/project-graph';
+import {
+  calculateProjectDependencies,
+  DependentBuildableProjectNode,
+} from '@nrwl/workspace/src/utilities/buildable-libs-utils';
+import { mkdirSync, readFileSync, writeFileSync } from 'fs';
 
 process.env.NODE_ENV ??= 'test';
 
+const createTmpJestConfig = (
+  jestConfigPath: string,
+  root: string,
+  context: ExecutorContext,
+  dependencies: DependentBuildableProjectNode[]
+): string => {
+  const jestConfig = readFileSync(
+    path.resolve(context.cwd, jestConfigPath)
+  ).toString();
+
+  const overrides = createJestOverrides(root, context, dependencies);
+
+  return `
+    const path = require('path');
+    const { pathsToModuleNameMapper } = require('ts-jest/utils')
+    ${jestConfig}
+    ${overrides}
+  `;
+};
+
+export function createJestOverrides(
+  root: string,
+  context: ExecutorContext,
+  dependencies: DependentBuildableProjectNode[]
+): string {
+  const offset = offsetFromRoot(root);
+
+  const tsAliasPrefix = `@${context.workspace.npmScope}/`;
+
+  const paths = dependencies
+    .filter((dep) => !dep.node.name.startsWith('npm:'))
+    .reduce((prev, dep) => {
+      const node = dep.node as ProjectGraphProjectNode;
+
+      const rootPath = node.data.root.split('/');
+      rootPath.shift();
+
+      const distPackageJson = readJsonFile(
+        path.join(context.cwd, 'dist', node.data.root, 'package.json')
+      );
+
+      const indexFile = distPackageJson['main'] ?? distPackageJson['esm2015'];
```

Comment:
Wasn't sure what the best way is to find the main index.js in the repo, I am assuming node libs have `main` and angular packaged libs will have either main or default to the esm2015